website/addons/box/model.py

```diff
@@ -0,0 +1,328 @@
+        The dictionary keys and values will be available as variables in
+        box_user_settings.mako.
+        """
+        output = super(BoxUserSettings, self).to_json(self.owner)
+        output['has_auth'] = self.has_auth
+        return output
+
+    def token_refreshed_callback(self, access_token, refresh_token):
+        self.access_token = access_token
+        self.refresh_token = refresh_token
+        self.last_refreshed = datetime.utcnow()
+        self.save()
+
+    @property
+    def has_auth(self):
+        return bool(self.access_token and refresh_creds_if_necessary(self))
+
+    def delete(self, save=True):
+        self.clear()
+        super(BoxUserSettings, self).delete(save)
+
+    def clear(self):
+        """Clear settings and deauthorize any associated nodes."""
+        self.box_id = None
+        self.access_token = None
+        for node_settings in self.boxnodesettings__authorized:
+            node_settings.deauthorize(Auth(self.owner))
+            node_settings.save()
+        return self
+
+    def get_credentialsv2(self):
+        return CredentialsV2(
+            self.access_token, 
+            self.refresh_token, 
+            settings.BOX_KEY,
+            settings.BOX_SECRET,
+            self.token_refreshed_callback,
+        )
+
+    def __repr__(self):
+        return u'<BoxUserSettings(user={self.owner.username!r})>'.format(self=self)
+
+
+class BoxNodeSettings(AddonNodeSettingsBase):
+
+    user_settings = fields.ForeignField(
+        'boxusersettings', backref='authorized'
+    )
+
+    folder = fields.StringField(default=None)
+    folder_id = fields.IntegerField(default=None)
+
+    #: Information saved at the time of registration
+    #: Note: This is unused right now
+    registration_data = fields.DictionaryField()
+
+
+    @property
+    def display_name(self):
+        return '{0}: {1}'.format(self.config.full_name, self.folder)
+
+    @property
+    def has_auth(self):
+        """Whether an access token is associated with this node."""
+        return bool(self.user_settings and self.user_settings.has_auth and refresh_creds_if_necessary(self.user_settings))
+
+    def set_folder(self, folder, folder_id, auth):
+        self.folder = folder
+        self.folder_id = folder_id
+        # Add log to node
+        nodelogger = BoxNodeLogger(node=self.owner, auth=auth)
+        nodelogger.log(action="folder_selected", save=True)
+
+    def set_user_auth(self, user_settings):
+        """Import a user's Box authentication and create a NodeLog.
+
+        :param BoxUserSettings user_settings: The user settings to link.
+        """
+        self.user_settings = user_settings
+        nodelogger = BoxNodeLogger(node=self.owner, auth=Auth(user_settings.owner))
+        nodelogger.log(action="node_authorized", save=True)
+
+    def find_or_create_file_guid(self, path):
+        return BoxFile.get_or_create(self.owner, clean_path(os.path.join(self.folder, path.lstrip('/'))))
```

Comment:
As box is a new addon you can just pass the path in here without any transformation.
The only reason path is mutated else where is to avoid have to do any form of migration.